discovery/db_discovery.go

```diff
@@ -0,0 +1,92 @@
+package discovery
+
+import (
+	"time"
+
+	"github.com/livepeer/go-livepeer/common"
+	"github.com/livepeer/go-livepeer/core"
+	lpTypes "github.com/livepeer/go-livepeer/eth/types"
+	"github.com/livepeer/go-livepeer/net"
+
+	"github.com/golang/glog"
+)
+
+type DBOrchestratorPoolCache struct {
+	node *core.LivepeerNode
+}
+
+func NewDBOrchestratorPoolCache(node *core.LivepeerNode) *DBOrchestratorPoolCache {
+	if node.Eth == nil {
+		glog.Error("Could not refresh DB list of orchestrators: LivepeerNode nil")
+		return nil
+	}
+
+	ticker := time.NewTicker(1 * time.Hour)
+	go func(node *core.LivepeerNode) {
+		for {
+			select {
+			case <-ticker.C:
+				orchestrators, err := node.Eth.RegisteredTranscoders()
+				if err != nil {
+					glog.Error("Could not refresh DB list of orchestrators: ", err)
+					return
+				}
+				_, dbOrchErr := CacheDBOrchs(node, orchestrators)
+				if dbOrchErr != nil {
+					glog.Error("Could not refresh DB list of orchestrators: CacheDBOrchs err")
+					return
+				}
+			}
+		}
+	}(node)
+
+	return &DBOrchestratorPoolCache{node: node}
+}
+
+func CacheDBOrchs(node *core.LivepeerNode, orchs []*lpTypes.Transcoder) ([]*common.DBOrch, error) {
+	var dbOrchs []*common.DBOrch
+	if orchs == nil {
+		glog.Error("No new DB orchestrators created: no orchestrators found onchain")
+		return dbOrchs, nil
+	}
+
+	for _, orch := range orchs {
```

Comment:
Do we need a `if orch == nil { continue }` check here? Only reason for asking is because EthOchToDBOrch has one, so not sure if there's a case where the `orchs` list could contain a nil entry.

Reply:
I added the check to `EthOchToDBOrch` to be safe. I'm not sure orchs from `orchestrators, err := node.Eth.RegisteredTranscoders()` can come back nil (i.e. if an error is thrown when `tAddr, err := c.GetFirstTranscoderInPool()` or `tAddr, err = c.GetNextTranscoderInPool(tAddr)` is nil). We can leave both checks in to be safe, though technically as things are written now, if we're checking for a nil orch in this for loop above, we don't need to check in `EthOrchToDBOrch`.